server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -232,8 +232,14 @@ private Set<SegmentIdWithShardSpec> getPendingSegmentsForIntervalWithHandle(
     final ResultIterator<byte[]> dbSegments =
         handle.createQuery(
             StringUtils.format(
-                "SELECT payload FROM %1$s WHERE dataSource = :dataSource AND start <= :end and %2$send%2$s >= :start",
-                dbTables.getPendingSegmentsTable(), connector.getQuoteString()
+                // This query might fail if the year has a different number of digits
+                // See https://github.com/apache/druid/pull/11582 for a similar issue
+                // Using long for these timestamps instead of varchar would give correct time comparisons
+                "SELECT start, %2$send%2$s, payload FROM %1$s "
```

Comment:
Nit: Do start and end still need to be selected in the query?

Reply:
Thanks for catching it, yeah, we can remove this now.